colmi_r02_client/real_time/enum.py

```diff
@@ -0,0 +1,46 @@
+from enum import IntEnum, StrEnum
+
+class Action(IntEnum):
+    START = 1
+    PAUSE = 2
+    CONTINUE = 3
+    STOP = 4
+
+class RealTimeReading(IntEnum):
+    """
+    Taken from https://colmi.puxtril.com/commands/#data-request
+    """
+    HEART_RATE = 1
+    BLOOD_PRESSURE = 2
+    SPO2 = 3
+    FATIGUE = 4
+    HEALTH_CHECK = 5
+    # leaving this out as it's redundant
+    # REAL_TIME_HEART_RATE = 6
+    ECG = 7
+    PRESSURE = 8
+    BLOOD_SUGAR = 9
+    HRV = 10
+
+class RealTimeName(StrEnum):
+    HEART_RATE = "heart-rate" # Works for R02
+    SPO2 = "spo2" # Works for R02
+    BLOOD_PRESSURE = "blood-pressure" # Works for R02 (not sure if it's accurate)
+    FATIGUE = "fatigue" # Works for R02 (not sure if it's accurate)
+    HEALTH_CHECK = "health-check" # Works for R02 (not sure if it's accurate)
+    ECG = "ecg" # Works for R02 (not sure if it's accurate)
+    PRESSURE = "pressure" # Works for R02 (not sure if it's accurate)
+    BLOOD_SUGAR = "blood-sugar" # Works for R02 (not sure if it's accurate)
+    HRV = "hrv" # Works for R02 (not sure if it's accurate)
+
+REAL_TIME_MAPPING = {
+    RealTimeName.HEART_RATE: RealTimeReading.HEART_RATE,
+    RealTimeName.BLOOD_PRESSURE: RealTimeReading.BLOOD_PRESSURE,
+    RealTimeName.SPO2: RealTimeReading.SPO2,
+    RealTimeName.FATIGUE: RealTimeReading.FATIGUE,
+    RealTimeName.HEALTH_CHECK: RealTimeReading.HEALTH_CHECK,
+    RealTimeName.ECG: RealTimeReading.ECG,
+    RealTimeName.PRESSURE: RealTimeReading.PRESSURE,
+    RealTimeName.BLOOD_SUGAR: RealTimeReading.BLOOD_SUGAR,
+    RealTimeName.HRV: RealTimeReading.HRV,
+}
```

Comment:
Why not just do away with the `RealTimeName` enum and just keep the dictionary instead? Something like:

```python
REAL_TIME_MAPPING = {
 'heart-rate': RealTimeReading.HEART_RATE,
  # etc...
}
```

Or better yet, you combine `RealTimeName` and `RealTimeReading` into a dataclass with a name and key attribute e.g.

```python
@dataclass(frozen=True)
class RealTimeReading:
  name: str
  key: int
```
^ Then you can use the same class in the CLI options and in `packet.py`.

Reply:
we could totally live without the `RealTimeName` enum. it was just a nice way of putting the names somewhere that kind of made sense but these just exist for the CLI pretty much, so yeah I agree it's a bit unnecessary. I'll remove it and do just the dict right away.

regarding combining them, it would make sense! i have some doubts though. how would you go about defining each type of reading? i feel like you would end up with some kind of list like
```python
REAL_TIME_READING_TYPES = (
    RealTimeReading(name='heart-rate', key=1),
    RealTimeReading(name='blood-pressure', key=2),
    RealTimeReading(name='spo2', key=3),
   # the rest of them...
)
```
and then for getting a specific reading by name you would have to go through the list and find the one with the name you want. Or you would have to go back to having a dict for quickly finding it. or doing something kinda dirty like:
```python
REAL_TIME_READING_TYPES_BY_NAME = {r.name: r for r in REAL_TIME_READING_TYPES}
```

and i don't know if the name itself is something relevant enough that we would want to keep it in a class to be reused in `packet.py`. The name is simply a way for the cli to use each reading but it's not important to the rest of the client

let me know what you think and thanks for the comments!

Reply:
I was thinking you could still keep the dict for mapping a CLI-friendly name to a class. Then you can use the dict whenever you need to get the specific class to pass to the functions in packet.py. No need to iterate over a list. Maybe I’m missing something?

Reply:
yes! agreed! I just don't see the point in having a class with both the `name` and `key`, when the name is only ever going to be useful for easy access to the cli, and it's going to be put in a dict with the names as keys either ways. so something like
```python
REAL_TIME_MAPPING = {
 'heart-rate': RealTimeReading(name="heart-rate", key=1),
  # etc...
}
```

seems to me like it'd be kind of redundant no? i'd rather have it as you first suggested:
```python
REAL_TIME_MAPPING = {
 'heart-rate': RealTimeReading.HEART_RATE,
  # etc...
}
```